### ISSUE TITLE
2018.2.14 docker dj

### ISSUE THREAD

**Issue by vieyahn2017** [vieyahn2017/IAmInPactera#41]
docker on dj

**Comment by vieyahn2017** [comment]
docker images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
dashboard           1.2.10.1            2eee4deb1173        3 weeks ago         1.01GB
oms-agent           1.2.10.1            a4ac39fb7b5a        3 weeks ago         383MB
oms-controller      1.2.10.1            65398c06d81b        3 weeks ago         383MB
dps                 1.2.10.1            24de46a1df47        3 weeks ago         603MB
hermes              1.2.10.1            98d1a0bba47e        3 weeks ago         595MB
heat                1.2.10.1            f0188718281d        3 weeks ago         543MB
topo                1.2.10.1            92da2c3dea91        3 weeks ago         399MB
host                1.2.10.1            21a58d585f66        3 weeks ago         534MB
cinder              1.2.10.1            9c38cae7ff88        3 weeks ago         703MB
bws                 1.2.10.1            d3210a4f1937        3 weeks ago         533MB
hyperfile-service   1.2.0.1             196a41fe91c9        3 weeks ago         462MB
hyperfile-api       1.2.0.1             a6599cbcd622        3 weeks ago         462MB
gollum              1.2.10.1            782af34fb8d8        3 weeks ago         394MB
drextend            1.2.10.1            d88e19c9a0d4        3 weeks ago         412MB
hydra               1.2.10.1            333a6a5c8fc6        3 weeks ago         418MB
medusa              1.2.10.1            64d832475120        3 weeks ago         603MB
filemeter-service   1.2.0.9             d61f05026d2f        3 weeks ago         463MB
filemeter-api       1.2.0.9             43269c0237e3        3 weeks ago         463MB
authkeepmgt         1.2.10.1            71033a335665        3 weeks ago         366MB
oms-api             1.2.10.1            87372e1e7ac8        3 weeks ago         383MB
dcms                1.2.10.1            cf97f6e50086        3 weeks ago         374MB
manila-share        1.2.0.10            df393f51b6a1        3 weeks ago         465MB
manila-scheduler    1.2.0.10            d20b7ec3337f        3 weeks ago         465MB
manila-api          1.2.0.10            273582c3f64e        3 weeks ago         465MB
certms              1.2.10.1            ab2aeead8564        3 weeks ago         374MB
llo                 1.2.10.1            51c63773d5eb        3 weeks ago         384MB
license             1.2.10.1            97b23a803459        3 weeks ago         814MB
keystone            1.2.10.1            60dfdcc1095d        3 weeks ago         405MB
fms                 1.2.10.1            9b4f8e076a63        3 weeks ago         532MB
rabbitmq            1.2.10.1            a8d09b19621b        3 weeks ago         555MB
cms                 1.2.10.1            8ae50d39fd13        3 weeks ago         376MB
haproxy             1.2.10.1            a540db10cd8c        3 weeks ago         315MB
zookeeper           1.2.10.1            43aefa78eaee        3 weeks ago         588MB
gaussdb             1.2.10.1            87c3e8292679        3 weeks ago         310MB
basejavaimage       006                 eb6ae68363e1        7 weeks ago         459MB
basepythonimage     006                 84c12a6d0a3e        8 weeks ago         331MB
baseimg             006                 2dfeedab4601        8 weeks ago         230MB

**Comment by vieyahn2017** [comment]
```shell
DJ01:~ # docker ps
CONTAINER ID        IMAGE                       COMMAND                  CREATED             STATUS              PORTS               NAMES
6bb08a3a29bb        rabbitmq:1.2.10.1           "bash /usr/local/l..."   2 days ago          Up 2 days                               rabbitmq
3004a77d0080        dashboard:1.2.10.1          "bash /etc/dashboa..."   3 weeks ago         Up 3 weeks                              dashboard
ff398bf96ef6        oms-agent:1.2.10.1          "/bin/bash /usr/bi..."   3 weeks ago         Up 3 weeks                              oms-agent
7f261c7e4b1d        oms-controller:1.2.10.1     "/bin/bash /usr/bi..."   3 weeks ago         Up 3 weeks                              oms-controller
e67c3afbbdcd        dps:1.2.10.1                "bash -c '/usr/loc..."   3 weeks ago         Up 3 weeks                              dps
98b44c0a44e8        hermes:1.2.10.1             "/bin/bash -c 'sh ..."   3 weeks ago         Up 3 weeks                              hermes
8e3f77020122        heat:1.2.10.1               "bash -c 'sh /inst..."   3 weeks ago         Up 5 days                               heat-engine
abee4144fab6        heat:1.2.10.1               "bash -c 'sh /inst..."   3 weeks ago         Up 3 weeks                              heat-api
a37196d24e27        topo:1.2.10.1               "bash /opt/topo/st..."   3 weeks ago         Up 3 weeks                              topo
ea006bfcc161        host:1.2.10.1               "bash -c 'sh /inst..."   3 weeks ago         Up 3 weeks                              host-polling
efaf7e19c261        host:1.2.10.1               "bash -c 'sh /inst..."   3 weeks ago         Up 3 weeks                              host-api
65cfe6a8a622        cinder:1.2.10.1             "bash -c '/usr/loc..."   3 weeks ago         Up 3 weeks                              cinder
daab539f1594        bws:1.2.10.1                "bash -c 'sh /inst..."   3 weeks ago         Up 3 weeks                              bws-api
89ced1be82ba        hyperfile-service:1.2.0.1   "/bin/bash /usr/bi..."   3 weeks ago         Up 3 weeks                              hyperfile-service
0e22facfff9f        hyperfile-api:1.2.0.1       "/bin/bash /usr/bi..."   3 weeks ago         Up 3 weeks                              hyperfile-api
1bdeffcc746c        gollum:1.2.10.1             "bash -c /opt/goll..."   3 weeks ago         Up 3 weeks                              gollum
93d339c0da6c        drextend:1.2.10.1           "bash -c '/usr/loc..."   3 weeks ago         Up 3 weeks                              drextend
fc220a4b7f8e        hydra:1.2.10.1              "bash /opt/hydra/s..."   3 weeks ago         Up 3 weeks                              hydra
2649c10f5914        medusa:1.2.10.1             "/bin/bash -c 'sh ..."   3 weeks ago         Up 3 weeks                              medusa
51940080d5c6        filemeter-service:1.2.0.9   "/bin/bash /usr/bi..."   3 weeks ago         Up 3 weeks                              filemeter-service
c1fd92aaecad        filemeter-api:1.2.0.9       "/bin/bash /usr/bi..."   3 weeks ago         Up 3 weeks                              filemeter-api
14ec933e220a        authkeepmgt:1.2.10.1        "/bin/bash -c /usr..."   3 weeks ago         Up 3 weeks                              authkeepmgt
615ac58c8fd3        oms-api:1.2.10.1            "/bin/bash /usr/bi..."   3 weeks ago         Up 2 days                               oms-api
c046a9e16713        dcms:1.2.10.1               "/bin/bash -c /usr..."   3 weeks ago         Up 3 weeks                              dcms
abde37c0bac7        manila-scheduler:1.2.0.10   "/bin/bash /usr/bi..."   3 weeks ago         Up 3 weeks                              manila-scheduler
b64e7415f7cb        manila-api:1.2.0.10         "/bin/bash /usr/bi..."   3 weeks ago         Up 3 weeks                              manila-api_tenant
40ad067a3b5d        manila-api:1.2.0.10         "/bin/bash /usr/bi..."   3 weeks ago         Up 3 weeks                              manila-api_admin
45ea5a74ef13        certms:1.2.10.1             "/bin/bash -c /usr..."   3 weeks ago         Up 3 weeks                              certms
8ed8edacad52        llo:1.2.10.1                "/bin/bash /usr/bi..."   3 weeks ago         Up 2 days                               llo
ac6a849fdc9f        license:1.2.10.1            "bash /opt/license..."   3 weeks ago         Up 3 weeks                              license
9bb88369fa26        keystone:1.2.10.1           "/bin/bash keyston..."   3 weeks ago         Up 3 weeks                              keystone
1db09042daf5        fms:1.2.10.1                "bash /opt/huawei/..."   3 weeks ago         Up 3 weeks                              fms
693c132d1fb5        cms:1.2.10.1                "bash /etc/cms/cms..."   3 weeks ago         Up 2 days                               cms
48b49257a22b        haproxy:1.2.10.1            "bash /etc/huawei/..."   3 weeks ago         Up 3 weeks                              haproxyinternal
127b413e4be5        zookeeper:1.2.10.1          "/bin/bash -c 'bas..."   3 weeks ago         Up 3 weeks                              zookeeper
3062354be605        gaussdb:1.2.10.1            "bash /home/start_..."   3 weeks ago         Up 3 weeks                              gaussdb
```

**Comment by vieyahn2017** [comment]
docker exec  -it  -u root dashboard bash

**Comment by vieyahn2017** [comment]
如何在Docker容器内外互相拷贝数据？
http://blog.csdn.net/yangzhenping/article/details/43667785

从容器内拷贝文件到主机上
[plain] view plain copy
docker cp <containerId>:/file/path/within/container /host/path/target  
 
从主机上拷贝文件到容器内

参考自：
http://stackoverflow.com/questions/22907231/copying-files-from-host-to-docker-container

1.用-v挂载主机数据卷到容器内

 
[plain] view plain copy
docker run -v /path/to/hostdir:/mnt $container  
在容器内拷贝  
cp /mnt/sourcefile /path/to/destfile  

 
 
2.直接在主机上拷贝到容器物理存储系统
 

A. 获取容器名称或者id :
[plain] view plain copy
$ docker ps  

 
 
B. 获取整个容器的id
[plain] view plain copy
$ docker inspect -f   '{{.Id}}'  步骤A获取的名称或者id  

 
 
C. 在主机上拷贝文件:
[plain] view plain copy
$ sudo cp path-file-host /var/lib/docker/aufs/mnt/FULL_CONTAINER_ID/PATH-NEW-FILE   
或者  
$ sudo cp path-file-host /var/lib/docker/devicemapper/mnt/123abc<<id>>/rootfs/root  

 
例子：
[plain] view plain copy
$ docker ps  
  
CONTAINER ID      IMAGE    COMMAND       CREATED      STATUS       PORTS        NAMES  
  
d8e703d7e303   solidleon/ssh:latest      /usr/sbin/sshd -D                      cranky_pare  
  
$ docker inspect -f   '{{.Id}}' cranky_pare  
  
or   
$ docker inspect -f   '{{.Id}}' d8e703d7e303  
  
d8e703d7e3039a6df6d01bd7fb58d1882e592a85059eb16c4b83cf91847f88e5  
  
$ sudo cp file.txt /var/lib/docker/aufs/mnt/**d8e703d7e3039a6df6d01bd7fb58d1882e592a85059eb16c4b83cf91847f88e5  

3.用输入输出符

 
[plain] view plain copy
docker run -i ubuntu /bin/bash -c 'cat > /path/to/container/file' < /path/to/host/file/  

或者
[plain] view plain copy
docker exec -it <container_id> bash -c 'cat > /path/to/container/file' < /path/to/host/file/

**Comment by vieyahn2017** [comment]
# docker inspect dashboard
```json
[
    {
        "Id": "731ef5512f83d810fb333148f3ebf41d559e1ac74039be62b82baabd7426f852",
        "Created": "2018-04-23T07:28:29.213733932Z",
        "Path": "bash",
        "Args": [
            "/etc/dashboard/start_dashboard.sh"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 779,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2018-04-23T07:28:30.073851037Z",
            "FinishedAt": "0001-01-01T00:00:00Z"
        },
        "Image": "sha256:883b77764a179ec9a7b0b6777dfd5cd880ae82829da02c2a13a0f459578e1354",
        "ResolvConfPath": "/opt/huawei/dj/docker/containers/731ef5512f83d810fb333148f3ebf41d559e1ac74039be62b82baabd7426f852/resolv.conf",
        "HostnamePath": "/opt/huawei/dj/docker/containers/731ef5512f83d810fb333148f3ebf41d559e1ac74039be62b82baabd7426f852/hostname",
        "HostsPath": "/opt/huawei/dj/docker/containers/731ef5512f83d810fb333148f3ebf41d559e1ac74039be62b82baabd7426f852/hosts",
        "LogPath": "/opt/huawei/dj/docker/containers/731ef5512f83d810fb333148f3ebf41d559e1ac74039be62b82baabd7426f852/731ef5512f83d810fb333148f3ebf41d559e1ac74039be62b82baabd7426f852-json.log",
        "Name": "/dashboard",
        "RestartCount": 0,
        "Driver": "devicemapper",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "/opt/huawei/dj/DJSecurity/privkey:/opt/huawei/dj/DJSecurity/privkey",
                "/opt/huawei/dj/DJSecurity/server-cert/cli-client:/opt/huawei/dj/DJSecurity/server-cert/cli-client",
                "/opt/omm/oma/tools:/opt/omm/oma/tools",
                "/var/log/huawei/dj/services/system/kmc:/var/log/huawei/dj/services/system/kmc",
                "/etc/localtime:/etc/localtime:ro",
                "/etc/hosts:/etc/temp_hosts:ro",
                "/opt/huawei/dj/DJSecurity/mgmt-dev-ca:/opt/huawei/dj/DJSecurity/mgmt-dev-ca",
                "/opt/huawei/dj/DJSecurity/server-cert/hermes:/opt/huawei/dj/DJSecurity/server-cert/hermes",
                "/var/log/huawei/dj/services/system/dashboard:/var/log/fusionsphere/component/dashboard",
                "/opt/huawei/dj/paas:/opt/huawei/dj/paas:ro",
                "/opt/huawei/dj/cfg:/opt/huawei/dj/cfg:ro",
                "/opt/huawei/dj/DJSecurity/server-ca:/opt/huawei/dj/DJSecurity/server-ca",
                "/opt/huawei/dj/DJSecurity/mgmt-dev-ca-bundle-files:/opt/huawei/dj/DJSecurity/mgmt-dev-ca-bundle-files",
                "/opt/huawei/dj/DJSecurity/server-ha-ca:/opt/huawei/dj/DJSecurity/server-ha-ca",
                "/opt/huawei/dj/DJSecurity/server-cert/dashboard:/opt/huawei/dj/DJSecurity/server-cert/dashboard",
                "/opt/fusionplatform/data/openstack:/opt/fusionplatform/data/openstack:ro"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "host",
            "PortBindings": {},
            "RestartPolicy": {
                "Name": "no",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Dns": [],
            "DnsOptions": [],
            "DnsSearch": [],
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "host",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 2048,
            "Memory": 4294967296,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": [],
            "DeviceCgroupRules": null,
            "DiskQuota": 0,
            "KernelMemory": 0,
            "MemoryReservation": 0,
            "MemorySwap": -1,
            "MemorySwappiness": -1,
            "OomKillDisable": false,
            "PidsLimit": 1024,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0
        },
        "GraphDriver": {
            "Data": {
                "DeviceId": "451",
                "DeviceName": "docker-8:3-403039585-4ef42e103606e17c0b44091e69b092cd1c312c823f8323926bf0149406ff5e4a",
                "DeviceSize": "10737418240"
            },
            "Name": "devicemapper"
        },
        "Mounts": [
            {
                "Type": "bind",
                "Source": "/opt/huawei/dj/DJSecurity/server-ha-ca",
                "Destination": "/opt/huawei/dj/DJSecurity/server-ha-ca",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/var/log/huawei/dj/services/system/kmc",
                "Destination": "/var/log/huawei/dj/services/system/kmc",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/huawei/dj/DJSecurity/mgmt-dev-ca",
                "Destination": "/opt/huawei/dj/DJSecurity/mgmt-dev-ca",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/huawei/dj/paas",
                "Destination": "/opt/huawei/dj/paas",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/huawei/dj/cfg",
                "Destination": "/opt/huawei/dj/cfg",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/huawei/dj/DJSecurity/server-ca",
                "Destination": "/opt/huawei/dj/DJSecurity/server-ca",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/fusionplatform/data/openstack",
                "Destination": "/opt/fusionplatform/data/openstack",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/huawei/dj/DJSecurity/server-cert/cli-client",
                "Destination": "/opt/huawei/dj/DJSecurity/server-cert/cli-client",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/omm/oma/tools",
                "Destination": "/opt/omm/oma/tools",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/huawei/dj/DJSecurity/server-cert/hermes",
                "Destination": "/opt/huawei/dj/DJSecurity/server-cert/hermes",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/var/log/huawei/dj/services/system/dashboard",
                "Destination": "/var/log/fusionsphere/component/dashboard",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/huawei/dj/DJSecurity/privkey",
                "Destination": "/opt/huawei/dj/DJSecurity/privkey",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/etc/hosts",
                "Destination": "/etc/temp_hosts",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/huawei/dj/DJSecurity/mgmt-dev-ca-bundle-files",
                "Destination": "/opt/huawei/dj/DJSecurity/mgmt-dev-ca-bundle-files",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/opt/huawei/dj/DJSecurity/server-cert/dashboard",
                "Destination": "/opt/huawei/dj/DJSecurity/server-cert/dashboard",
                "Mode": "",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/etc/localtime",
                "Destination": "/etc/localtime",
                "Mode": "ro",
                "RW": false,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "DJ01",
            "Domainname": "",
            "User": "openstack",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "cms_ip=8.46.196.187",
                "internal_ip=191.46.196.181",
                "gaussdb_float_ip=191.46.196.100",
                "tenant_float_ip=8.46.196.188",
                "internal_ip_list=191.46.196.181,191.46.196.183,191.46.196.185",
                "profile=dj",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
            ],
            "Cmd": [
                "bash",
                "/etc/dashboard/start_dashboard.sh"
            ],
            "Image": "dashboard:1.2.10.2",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {}
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "1439098bfbd7a4679a51e1f09c85bfa9ae3fb9b39acc9391c958d151e0f5ef7d",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/default",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "host": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "1141e1b75e79abeb44ae480e45c0cae6e47c4655807b4ebe54d50b9fcc596d64",
                    "EndpointID": "bb8534ab2cc9d7d2c09adabefe4115e73d3ae9b72d754fe77b2caba721894d55",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]
```

**Comment by vieyahn2017** [comment]
docker info
Containers: 38
 Running: 37
 Paused: 0
 Stopped: 1
Images: 126
Server Version: 17.06.0-ce
Storage Driver: devicemapper
 Pool Name: docker-8:3-403039585-pool
 Pool Blocksize: 65.54kB
 Base Device Size: 10.74GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 7.894GB
 Data Space Total: 107.4GB
 Data Space Available: 99.48GB
 Metadata Space Used: 15.63MB
 Metadata Space Total: 2.147GB
 Metadata Space Available: 2.132GB
 Thin Pool Minimum Free Space: 10.74GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /opt/huawei/dj/docker/devicemapper/devicemapper/data
 Metadata loop file: /opt/huawei/dj/docker/devicemapper/devicemapper/metadata
 Library Version: 1.03.01 (2015-05-15)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version:  (expected: cfb82a876ecc11b5ca0977d1733adbe58599088a)
runc version: N/A (expected: 2d41c047c83e09a6d61d464906feb2a2f3c52aa4)
init version: v0.13.0 (expected: 949e6facb77383876aeff8a6944dde66b3089574)
Security Options:
 apparmor
Kernel Version: 4.4.21-69-default
Operating System: SUSE Linux Enterprise Server 12 SP2
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.67GiB
Name: DJ01
ID: JKCZ:V65J:DBM3:WSWD:GC2V:SYQC:TN2K:EJRO:ZQDO:6LPH:SJVL:OG2X
Docker Root Dir: /opt/huawei/dj/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: true

WARNING: devicemapper: usage of loopback devices is strongly discouraged for production use.
         Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
WARNING: No swap limit support
WARNING: No kernel memory limit support

**Comment by vieyahn2017** [comment]
cat /proc/version
Linux version 4.4.21-69-default (geeko@buildhost) (gcc version 4.8.5 (SUSE Linux) ) #1 SMP Tue Oct 25 10:58:20 UTC 2016 (9464f67)
